### PR TITLE
New `failing test` management using Jupiter extension

### PR DESCRIPTION
**Pull request by JooHyukKim**
This is from discussion regarding https://github.com/FasterXML/jackson/discussions/243#discussioncomment-10620791.

### Descrption

`jackson-databind` seems like reasonable start since it is extended by most repos.

On approval will do the following for closure ...
- move failing tests to where they belong(?) main `/test` dir
- get rid of related maven config

### Resources

- [JUnit Extension Doc](https://junit.org/junit5/docs/current/user-guide/#extensions-intercepting-invocations)

### CODE REVIEW

**Comment by cowtowncoder** [comment]
I think I like the idea of using annotation.

The one last thing I feel uneasy about is the location: while in some ways it is good to keep tests in their logical place (where they'd be if passing), it makes it more difficult to see the set of expected-to-fail cases. So...

* We cannot leave tests under `src/test/java/.../failing` since Maven settings in parent `pom.xml` skip these (I mean, we could change it etc but easier not to try that)
* But should there be something like `src/test/java/.../tofix` package? (sibling to existing "failing" package) Just a naming convention, no special handling, to contain test classes where at least some test methods are expected to fail?

WDYT?

I just like the ability to quickly see all tests that are for things-to-fix.

**Comment by JooHyukKim** [comment]
Yeah it makes sense since when any test start passing, we need to further take actions.

'tofix' would suffice 😆

File: src/test/java/com/fasterxml/jackson/databind/testutil/failure/JacksonTestFailureExpected.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.fasterxml.jackson.databind.testutil.failure;
+
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target({ElementType.METHOD})
```
Comment (cowtowncoder) [inline src/test/java/com/fasterxml/jackson/databind/testutil/failure/JacksonTestFailureExpected.java:10]: Could use Javadoc

**Review by cowtowncoder — APPROVED** [review]
Looks good -- but will probably wait until 2.19 branch started, after 2.18.0 release in a week or so.

**Comment by cowtowncoder** [comment]
Re-thinking this one: probably want to do for 2.18. Just need to see how to get things merged best way.

**Review by cowtowncoder — APPROVED** [review]

**Comment by cowtowncoder** [comment]
Merged 3 new classes separately, went cleanly. Now tackling renaming which may get messier.

**Comment by cowtowncoder** [comment]
Phew! That merge resolution was quite something. But this is now fully merged, for 2.18 and 3.0!

**Comment by JooHyukKim** [comment]
> Phew! That merge resolution was quite something. But this is now fully merged, for 2.18 and 3.0!
> 
> 

Awww, thank u! 🙏🏽 Had I realized the work load, I would've done the merging into master. 

Please lemme know next time hehe @cowtowncoder

**Comment by cowtowncoder** [comment]
@JooHyukKim Sure, not a problem. But I was in the best position to do the merge since I had renamed couple of tests, knew 2 pass in 3.0 but not 2.x... lots of trivia needed :)
Those 2 cases were actually useful in validating that "passing fails annotated tests".

Also: I noticed there was one test (for issue #639 I think) duplicated so was able to delete that.

**Pull request merged**